pandas/io/gbq.py

```diff
@@ -160,7 +160,48 @@ def get_credentials(self):
         if self.private_key:
             return self.get_service_account_credentials()
         else:
-            return self.get_user_account_credentials()
+            # Try to retrieve Application Default Credentials
+            credentials = self.get_application_default_credentials()
+            if not credentials:
+                credentials = self.get_user_account_credentials()
+            return credentials
+
+    def get_application_default_credentials(self):
+        """
+        This method tries to retrieve the "default application credentials"
+        Could be useful for running code on Google Cloud Platform
+        """
+        oauth2client_library_imported = False
+        try:
+            from oauth2client.client import GoogleCredentials
+            from oauth2client.client import AccessTokenRefreshError
+            from oauth2client.client import HttpAccessTokenRefreshError
+            from oauth2client.client import ApplicationDefaultCredentialsError
+            oauth2client_library_imported = True
+            from googleapiclient.discovery import build
+            from googleapiclient.errors import HttpError
+        except ImportError:
+            if not oauth2client_library_imported:
+                return None
+            from apiclient.discovery import build
+            from apiclient.errors import HttpError
+
+        credentials = None
```

Comment:
you don't need to set credentials, ether you have them or the except has hit and you have returned